pyproject.toml

```diff
@@ -1,6 +1,260 @@
+authors = [{name = "The AiiDA team", email = "developers@aiida.net"}]
+readme = "README.md"
+license = {file = "LICENSE.txt"}
+classifiers = [
+        "Development Status :: 5 - Production/Stable",
+        "Framework :: AiiDA",
+        "License :: OSI Approved :: MIT License",
+        "Operating System :: POSIX :: Linux",
+        "Operating System :: MacOS :: MacOS X",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Topic :: Scientific/Engineering"
+]
+keywords = ["aiida", "workflows"]
+requires-python = ">=3.8"
+dependencies = [
+        "aldjemy~=2.4",
+        "alembic~=1.2",
+        "archive-path~=0.3.6",
+        "aio-pika~=6.6",
+        "circus~=0.17.1",
+        "click-config-file~=0.6.0",
+        "click-spinner~=0.1.8",
+        "click~=8.0,>=8.0.3",
+        "disk-objectstore~=0.6.0",
+        "django~=2.2",
+        "graphviz~=0.13",
+        "ipython~=7.20",
+        "jinja2~=3.0",
+        "jsonschema~=3.0",
+        "kiwipy[rmq]~=0.7.4",
+        "importlib-metadata~=4.3",
+        "numpy~=1.19",
+        "pamqp~=2.3",
+        "paramiko~=2.7,>=2.7.2",
+        "plumpy~=0.20.0",
+        "pgsu~=0.2.0",
+        "psutil~=5.6",
+        "psycopg2-binary~=2.8.3",
+        "python-dateutil~=2.8",
+        "pytz~=2021.1",
+        "pyyaml~=5.4",
+        "simplejson~=3.16",
+        "sqlalchemy~=1.4.22",
+        "tabulate~=0.8.5",
+        "typing-extensions; python_version < '3.8'",
+        "tqdm~=4.45",
+        "tzlocal~=2.0",
+        "upf_to_json~=0.9.2",
+        "wrapt~=1.11.1"
+]
+
+[project.urls]
+Home = "http://www.aiida.net/"
+Documentation = "https://aiida.readthedocs.io"
+Source = "https://github.com/aiidateam/aiida-core"
+
+[project.optional-dependencies]
+ssh_kerberos = [
+    "gssapi~=1.6",
+    "pyasn1~=0.4.8"
+]
+rest = [
+    "flask-cors~=3.0",
+    "flask-restful~=0.3.7",
+    "flask~=1.1",
+    "pyparsing~=2.4",
+    "python-memcached~=1.59",
+    "seekpath~=1.9,>=1.9.3"
+]
+docs = [
+    "docutils==0.15.2",
+    "pygments~=2.5",
+    "pydata-sphinx-theme~=0.6.3",
+    "sphinx~=3.2.1",
+    "sphinxcontrib-details-directive~=0.1.0",
+    "sphinx-panels~=0.5.0",
+    "sphinx-copybutton~=0.3.0",
+    "sphinx-notfound-page~=0.5",
+    "sphinxext-rediraffe~=0.2.4",
+    "sphinx-sqlalchemy~=0.1.1"
+]
+atomic_tools = [
+    "PyCifRW~=4.4",
+    "ase~=3.18",
+    "matplotlib~=3.3,>=3.3.4",
+    "pymatgen>=2019.7.2,<=2022.1.9,!=2019.9.7",
+    "pymysql~=0.9.3",
+    "seekpath~=1.9,>=1.9.3",
+    "spglib~=1.14"
+]
+notebook = [
+    "jupyter-client~=6.1,<6.1.13",
+    "jupyter~=1.0",
+    "notebook~=6.1,>=6.1.5"
+]
+pre-commit = [
+    "mypy==0.930",
+    "packaging==20.3",
+    "pre-commit~=2.2",
+    "pylint~=2.11.1",
+    "pylint-aiida~=0.1.1",
+    "pylint-django",
+    "sqlalchemy[mypy]~=1.4.29",
+    "tomli",
+]
```

Comment:
Might be out of scope for this PR, but does it really make sense to specify the pre-commit dependencies here? I find the workflow where pre-commit just installs the pre-commit dependencies into a dedicated environment much more convenient and sensible.

Reply:
so do I, but the problem at present is that we have a number of local hooks running, namely `pylint` and `mypy`.
To make these non-local, you would have to copy all of e.g. `requirements/requirements-py-3.8.txt` to under the `additional_dependencies` key in the pre-commit config file for both of these hooks, and then keep it up-to-date manually.


I would love to find a solution for this, but see e.g. https://github.com/pre-commit/pre-commit/issues/730 why it is not possible to get the requirements directly from the requirements file.